rfcs/0114-nixos-code-of-conduct.md

```diff
@@ -0,0 +1,108 @@
+you want to experiment with, make a fork and see how it works.
+- We will exclude you from interaction if you insult, demean or harass anyone.
+That is not welcome behavior. We interpret the term “harassment” as including the definition in the
+[Citizen Code of Conduct](https://github.com/stumpsyn/policies/blob/master/citizen_code_of_conduct.md);
+if you have any lack of clarity about what might be included in that concept,
+please read their definition. In particular, we don’t tolerate behavior that excludes
+people in socially marginalized groups.
+- Private harassment is also unacceptable. No matter who you are, if you feel
+you have been or are being harassed or made uncomfortable by a community member,
+please contact one of the channel ops or any of the NixOS moderation team immediately.
+Whether you’re a regular contributor or a newcomer, we care about making this community
+a safe place for you and we’ve got your back.
+- Likewise any spamming, trolling, flaming, baiting or other attention-stealing behavior is not welcome.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+There have been a few spectacular failures of communication and good faith within the community.
+Most notable recent examples include the [block evasion thread](https://discourse.nixos.org/t/github-block-evasion-is-not-acceptable/12763),
+heated discussions in [RFC #98](https://github.com/NixOS/rfcs/pull/98) and [RFC #111](https://github.com/NixOS/rfcs/pull/111).
+Each of these incidents have caused significant discussion on IRC (when it was still official),
+matrix, discourse, and would bleed over into github and even non-official platforms.
+These discussions are generally very polarizing, and causes an enormous amount
+of emotional and mental stress to those involved.
+These incidents are also very embarressing for the greater nix community,
+and not having a way to identify disruptive before it becomes a heated
+issue is detrimental to the health of the community.
+
+In conjunction with a moderation team (out-of-scope for this RFC), these incidents could have been
+better arbitrated as to have a more satisfactory resolution before escalation. This
+Code of Conduct will better equip the moderation process by providing clear expectations
+for behavior within the community.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- There are more "rules" for the community to follow. However, these statutes are intended to
+still allow individuals to participate within the community in a healthy way for all.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- #98 Also provides an opinionated values and goals for the moderation team. However,
+these rules are less well-defined, and allow for broader interpretations
+of unacceptable behavior.
+
+- #102 Codifies some of the existing moderation practices, and uses the
+existing [community ethos](https://nixos.org/community/index.html) as moderation criteria.
```

Comment:
Is #102 actually an alternative? I would say, keep only having community ethos as the guidelines — with or without #102 or any alternative to it. And I think #102 does not discourage defining the expectations in a separate document.

Reply:
Yes, but it's not specific about behaviors which have been causing issues.

102 isn't mutually exclusive with this PR. It could be viewed as supplemental.

Reply:
Yes, I just think the alternative to mention here is «Continue using just the statement of community ethos with no more detailed guidelines.», not something about #102

Reply:
Ah, fair.

Reply:
added